src/docs/arch-schema.dox

```diff
@@ -1,11 +1,117 @@
 /*! @arch_page arch-schema Schema
 
-A schema defines the format of the application data in WiredTiger. WiredTiger
-supports various types of schemas (See @ref schema for more
-information), operated upon through a WT_SESSION reference.
+A schema defines the format of the application data and how it will be stored by WiredTiger. While
+many tables have simple key/value pairs for records, WiredTiger also supports more complex data
+patterns. See @ref schema for more information.
 
-@subpage arch-schema-ops
+@section schema_data_formats Data Formats
 
-This section details the internals of various API operations that operate on schemas, such as WT_SESSION::create and WT_SESSION::rename.
+The format of keys and values is configured through \c key_format and \c value_format entries in
+@ref config_strings. WiredTiger supports simple or composite data formats for keys and values.
+See @ref schema_format_types for the full list of supported data types.
+ - A simple format stores data in one type, for example \c "key_format=i,value_format=S".
+ - A composite format can store multiple data types as a single blob. The configuration string
+ can be something like \c "key_format=Si,value_format=ul". Cursors support encoding and decoding of
+ these types of keys and values. See @ref arch_cursor_raw and @ref schema_cursor_formats for
+ more details.
 
+Column store requires the key format to be defined as the record number \c 'r' type.
+@ref schema has more information on key/value formats.
+
+@section schema_files Data Files
+
+Database schema defines how data files are organized in the database home folder:
+ - A row-oriented table keeps all the data in one file called \c "<table name>.wt", where
+ \c "<table name>" is the name that was passed as a part of \c name parameter to WT_SESSION::create.
+ - A column-oriented table stores the data in multiple files. One for each column group. Filename
```

Comment:
Minor nit: "The filename will be as follows..."

Reply:
Done